### PR TITLE
Added tests for reordering/denying extensions for Feature/new ws extension api

### PR DESCRIPTION
**Pull request by dpwspoon**

### CODE REVIEW

File: specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.partial.agreement/handshake.request.rpt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+connect http://localhost:8080/path?query
```
Comment (cmebarrow) [inline specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.partial.agreement/handshake.request.rpt]: What does partial agreement mean when there are no extensions requested?

File: specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.reordered/handshake.response.rpt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+connected
+
+read method "GET"
+read version "HTTP/1.1"
+read header "Host" "localhost:8080"
+read header "Upgrade" /(?i:websocket)/
+read header "Connection" /(?i:Upgrade)/
+read header "Sec-WebSocket-Key" /(?<key>[a-zA-Z0-9+\/=]{24})/
+read header "Sec-WebSocket-Version" "13"
+read header "Sec-WebSocket-Protocol" "primary, secondary"
```
Comment (cmebarrow) [inline specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.reordered/handshake.response.rpt]: According to [RFC-6455](http://tools.ietf.org/html/rfc6455#page-48) the header is 
Sec-WebSocket-Extensions

File: specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.partial.agreement/handshake.request.rpt (hunk 1 above)
Comment (cmebarrow) [inline specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.partial.agreement/handshake.request.rpt]: Oh I see, it's just that you're using the Sec-WebSocket-Protocol header. But should be Sec-WebSocket-Extensions  according to the RFC(see my next comment).

**Comment by cmebarrow** [comment]
@dpwspoon Reviewed- ok except for extensions header name

**Comment by dpwspoon** [comment]
ping @cmebarrow, this should all be fixed

File: specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.partial.agreement/handshake.request.rpt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+connected
+
+write method "GET"
+write version "HTTP/1.1"
+write header "Host" "localhost:8080"
+write header "Upgrade" "websocket"
+write header "Connection" "Upgrade"
+write header "Sec-WebSocket-Key" "dGhlIHNhbXBsZSBub25jZQ=="
+write header "Sec-WebSocket-Version" "13"
+write header "Sec-WebSocket-Protocol" "valid, invalid"
```
Comment (cmebarrow) [inline specification/ws/src/main/scripts/org/kaazing/specification/ws/opening/response.header.sec.websocket.extensions.partial.agreement/handshake.request.rpt]: Need to correct the write headers too

**Pull request merged**